### PR TITLE
Fix for Issue #107 - Bintray shutdown

### PR DESCRIPTION
**Pull request by brianbolze**
Hey folks. My team at [Core](https://github.com/hello-core) has a potential fix for the Bintray shutdown #107 . It's working for us and would love to give back to the team who worked hard on putting `koma` together in the first place. 

@kyonifer Let us know if this works!

### CODE REVIEW

**Comment by kyonifer** [comment]
Thanks for putting this together. Looks like a good solution to me-- host on a github maven repo. We might also want to update the documentation to show people the correct `dependencies` block to bring in koma from the GH maven repo. Do you happen to have a `dependencies` declaration you could share?

File: gradle.properties [hunk 1]
```diff
@@ -5,15 +5,17 @@ kotlin.code.style=official
 
 ### Publishing
 
+# NOTE: Can be set back to kyonifer after merging 
+
 # Set up your maven coordinates here, artifactId is defined per project.
-version = 0.12
-group = com.kyonifer
+version = 0.12.1
+group = com.core
```
Comment (kyonifer) [inline gradle.properties:12]: Presumably this should read `com.hellocore`, as that is the domain you own correct? We could also just namespace this at com.kyonifer still, either way works for me.

**Comment by kyonifer** [comment]
I agree that having them in a centralized place would be better. No need for you guys to trouble yourselves with that-- I already pushed them to kyonifer/koma, and I can merge this branch (+ namespace changes) if this route works out.

Looking into how to actually give users access to these maven artifacts: It looks like there's currently no way to do unauthenticated github maven access ([1](https://github.community/t/how-to-allow-unauthorised-read-access-to-github-packages-maven-repository/115517/8), [2](https://github.community/t/download-from-github-package-registry-without-authentication/14407/104)), and Github employees seem to officially support [checking in a PAT](https://github.community/t/download-from-github-package-registry-without-authentication/14407/38) to user repositories, intentionally obfuscating the encoding to work around GH's detection mechanism that disables PATs that have been checked in somewhere public. [This gentleman](https://developerlife.com/2021/02/06/publish-kotlin-library-as-gradle-dep/#generate-the-personal-access-tokens-that-will-be-needed-to-publish-and-import) has written a guide on PATs,  but in my limited testing, directly commiting the PAT into source the way he did will get the PAT deactivated (obfuscation is needed).

Alternatively, [this repo](https://github.com/jcansdale-test/maven-consume) has a gradle plugin that attempts to do that sort of thing automatically, by using a global public PAT. I was able to get a test project configured with the following build.gradle to download koma:

```
buildscript {
  ext.kotlin_version = '1.3.10'
  repositories {
    maven {
      url "https://plugins.gradle.org/m2/"
    }
    mavenCentral()
  }
  dependencies {
    classpath "io.github.0ffz:gpr-for-gradle:1.2.1"
    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
  }
}
apply plugin: "io.github.0ffz.github-packages"
apply plugin: 'kotlin'
apply plugin: 'application'

mainClassName = 'demo.HelloWorldKt'

defaultTasks 'run'

repositories {
  mavenCentral()
  repositories {
    maven githubPackage.invoke("kyonifer")
  }
}

dependencies {
  compile group: "com.kyonifer", name:"koma-core-ejml", version: "0.12.1"
  compile group: "com.kyonifer", name:"koma-plotting", version: "0.12.1"

  compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
  testCompile 'junit:junit:4.11'
  testCompile "org.jetbrains.kotlin:kotlin-test-junit:$kotlin_version"
}
```

Anyone know of a better solution? All of these options seem pretty hacky.

**Comment by drmoose** [comment]
This solution doesn't appear to include `-common` artifacts. At least none of the packages I can find seem to provide API symbols for the "common" project in a multiplatform build. Is that deliberate or am I missing something?

**Comment by drmoose** [comment]
I was indeed missing something. The common symbols are in the `-metadata` artifacts now. There seem to be a bunch of lingering issues with transitive dependencies but I can open a separate PR for that once I get it figured out more. 

:+1:

**Comment by kyonifer** [comment]
Going to go ahead and merge this, since it looks like a solution thats working for now.

**Pull request merged**